common/setup.c

```diff
@@ -213,9 +216,6 @@ void __noreturn __text_init kernel_start(uint32_t multiboot_magic, unsigned long
     /* Initialize Programmable Interrupt Controller */
     init_pic();
 
-    /* PIC is initialized - enable local interrupts */
-    sti();
```

Comment:
Ahh, I see ... you moved the enabling down after init_timers().
What's the reason for that, beside the lack of flags saving interrupt enable/disable functions?

Reply:
Main reason was to enable the interrupts when they can be actually useful. Right now the earlier moment when they can be useful is after we initialize IOAPIC and timers, hence the move.